build.gradle.kts

```diff
@@ -272,26 +273,36 @@ tasks.withType(Sign::class).configureEach {
     notCompatibleWithConfigurationCache("$name task does not support configuration caching")
 }
 
-fun gitHubReleaseName(): String {
-    if (isDevelopmentRelease) {
-        return "Development Build"
+fun gitHubReleaseName(): Provider<String> {
+    return if (isDevelopmentRelease) {
+        provider { "Development Build" }
     } else {
-        return version.toString()
+        appVersion
     }
 }
```

Comment:
I _think_ this could be shortened to `appVersion.map { isDevelopmentRelease ? "Development Build" : it}`
Similarly for the other two methods gitReleaseTag and distributionVersion.

Please give it a try and also verify CC-compatibility.

Reply:
Is one of these the preferred style?

Reply:
and would there be any reason not to apply that to the other similar functions?
```
fun gitHubReleaseName() = appVersion.map { if (isDevelopmentRelease) "Development release" else it }

fun gitReleaseTag() = appVersion.map { if (isDevelopmentRelease) "development-latest" else "v$it" }

fun distributionVersion() = appVersion.map { if (isDevelopmentRelease) "dev" else it}
```

Reply:
For small functions, the expression body (the second one) is more idiomatic Kotlin.

But, ultimately it is personal preference. Sometimes I will choose the former for consistency because its the only function that could be turned use an expression body.

I will leave it between you and @etiennestuder.

Reply:
Just pushed a commit with the code snippet I posted above

Reply:
> Is one of these the preferred style?

I would have chosen the style that declares the return type and is also what the other methods use. I know it might not be as concise as it can be, so in the end, it is just my personal preference for expressiveness over conciseness.

```
fun xyz: Provider<String> {
```

Reply:
> Similarly for the other two methods gitReleaseTag and distributionVersion.

Whoops, must have glanced over that, sorry.

> `fun xyz: Provider<String> {`

Yeah I agree, that style is clearer. Not having much Kotlin experience, I just thought to go with what might be considered the more idiomatic approach. I'll update to use this style, as it's both and clearer and more consistent with the other functions.

Reply:
Thanks, Tyler.

Reply:
@tylerbertrand Feel free to merge once you and @erichaagdev are happy with it.

Reply:
Sounds good - thanks @etiennestuder!